src/sage/matrix/matrix_space.py

```diff
@@ -434,9 +434,75 @@ def get_matrix_class(R, nrows, ncols, sparse, implementation):
 
 class MatrixSpace(UniqueRepresentation, Parent):
     """
-    The space of matrices of given size and base ring
+    The space of matrices of given size and base ring.
 
-    EXAMPLES:
+    INPUT:
+
+    - ``base_ring`` -- a ring
+
+    - ``nrows`` or ``row_keys`` -- (nonnegative integer) the number of rows, or
+      a finite or enumerated family of arbitrary objects that index the rows
+      of the matrix
+
+    - ``ncols`` or ``column_keys`` -- (nonnegative integer, default ``nrows``)
+      the number of columns, or a finite or enumerated family of arbitrary objects
```

Comment:
I would prefer to remove "or enumerated" for safety.

Reply:
OK, that's finite with me, done in 6885c2d